core/src/main/scala/org/apache/spark/util/JsonProtocol.scala

```diff
@@ -757,15 +757,15 @@ private[spark] object JsonProtocol {
 
   def taskResourceRequestMapFromJson(json: JValue): Map[String, TaskResourceRequest] = {
     val jsonFields = json.asInstanceOf[JObject].obj
-    jsonFields.map { case JField(k, v) =>
+    jsonFields.collect { case JField(k, v) =>
```

Comment:
Ah, I see.  The compilation of `map` in this case fails in v2.13.4?

Reply:
Yes.